src/world/Explosion.php

```diff
@@ -198,8 +235,9 @@ public function explodeB() : bool{
 
 		$air = VanillaItems::AIR();
 		$airBlock = VanillaBlocks::AIR();
+		$fireBlock = VanillaBlocks::FIRE();
 
-		foreach($this->affectedBlocks as $block){
+		foreach(Utils::promoteKeys($this->affectedBlocks) as $hash => $block){
```

Comment:
This shouldn't be needed, `$hash` is an integer

You probably ought to add `@phpstan-var array<int, Block>` to the property declaration